### DISCUSSION TITLE
Error Joining Phoenix Channel/ CORS request did not succeed

### DISCUSSION THREAD

**Discussion by matthewbcool** [mozilla/hubs#3421] [Hubs Cloud Questions] [answered]
Started from new fork, cloned repo and set up as recommended in docs (AWS, recipe one, personal).

I've tried logging back in and out `npm run login/logout`, shutting down and turning the stack on, and editing CORS and CSP settings. Phoenix Socket URL appears to be correct. I've also deleted and recreated the stack. What am I missing?

UPDATE: I'm now just getting cors errors: ‘Access-Control-Allow-Origin’ missing - I've whitelisted my domain and link domain. Still not able to run locally. I did not encounter this issue the previous three times I have stepped through this process. Wondering what is different this time around and not sure where to look.

**Comment by robinkwilson** [comment, marked as answer]
Quick clarifying question: **Are you trying to solve how to successfully run your stack locally against your backend with `npm start` (after `npm run login`)?**

Re: ‘Access-Control-Allow-Origin’ missing. 
On **your response on the endpoint** you're sending a request, you need to send the Access-Control-Allow-Origin header in the **response**. 

If you use node/express:
```javascript
// So all responses you send back have the correct headers to send to your hub
app.use((req, res, next) => {
            res.header('Access-Control-Allow-Origin', "*");
            // ... other headers
            next();
}) 
```

**Reply by matthewbcool** [reply]
Are you trying to solve how to successfully run your stack locally against your backend with npm start (after npm run login)?
Correct! 

I just tried to run npm start again without making any edits to webpack config/cors and I'm now getting a new error and no longer getting cors errors:

<img width="600" alt="Screen Shot 2020-12-01 at 11 31 19 AM" src="https://user-images.githubusercontent.com/4493657/100787694-e0a02700-33c8-11eb-9356-87c1963eb7b7.png">

**Reply by robinkwilson** [reply]
Ah yes! I've seen this error!

Try deleting your local storage in your browser.

**Reply by matthewbcool** [reply]
Welp, the old deleting local storage seems to have solved the issue. I have not made any other adjustments and I'm hacking away on hubs cloud again. Thanks @robinkwilson 

Wonder if the previous issues could have been related to the aws outages in some way from last week maybe?
